### PR TITLE
HPCC-31451 Unversioned WSDL change in ESP services

### PR DESCRIPTION
**Pull request by kenrowland**
Removed enum TpMachineType since code was expecting a string type. 
Fixed misspelled type from strint to string in ws_smc service definition.

Signed-Off-By: Kenneth Rowland kenneth.rowland@lexisnexisrisk.com

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by rpastrana — APPROVED** [review]
@kenrowland this seems fine, but also want @wangkx to review.

File: esp/scm/ws_topology.ecm [hunk 1]
```diff
@@ -415,14 +415,10 @@ ESPresponse [exceptions_inline,encode(0)] TpClusterInfoResponse
...
 
-ESPenum TpMachineType : int
-{
-    TPMACHINECLUSTER(0, "TPMACHINECLUSTER")
-};
 ESPrequest [] TpMachineQueryRequest
 {
     //Type of Node the parent is....
-    TpMachineType Type;
+    string        Type;
```
Comment (wangkx) [inline esp/scm/ws_topology.ecm:421]: The string works. But, I am wondering whether it should be a string ESPenum. Like:
ESPenum TpMachineType : string
{
        THORMACHINES("THORMACHINES"),
        ROXIEMACHINES("ROXIEMACHINES"),
        ...
};
For backward compatible, you may version out the '[depr_ver("?")]  string        Type;' and add new '[min_ver("?")] ESPenum TpMachineType MachineType;'.

**Review by wangkx — COMMENTED** [review]
@kenrowland one comment

File: esp/scm/ws_topology.ecm (hunk 1 above)
Comment (kenrowland) [reply esp/scm/ws_topology.ecm:421]: Looking at the method void CTpWrapper::getClusterMachineList, the second parameter is the Type field. That method compares to a considerable number of different values that are not defined, but are literal string compares. I believe changing to a string enum would require additional code changes to ensure all usages of _Type_ used values defined by the enum.

Comment (wangkx) [reply esp/scm/ws_topology.ecm:421]: @kenrowland has added https://track.hpccsystems.com/browse/HPCC-31458.

**Review by wangkx — APPROVED** [review]
@kenrowland I approved

**Comment by kenrowland** [comment]
@ghalliday Please merge

**Pull request merged**